packages/manager/apps/pci-kubernetes/src/hooks/useFlavors.ts

```diff
@@ -0,0 +1,194 @@
+    isKubeFlavorsPending ||
+    isCatalogPending ||
+    isAvailabilityPending ||
+    isQuotaPending;
+
+  const mergedFlavors = useMemo(() => {
+    if (!flavors || !kubeFlavors || !catalog || !availability || !quota)
+      return [];
+    const kubeFlavorNames = new Set(
+      kubeFlavors.map(({ name }) => name.replace('-flex', '')),
+    );
+    const result = [...kubeFlavorNames].map((name) => ({
+      ...flavors.find((_flavor) => _flavor.name === name),
+      ...kubeFlavors.find((_flavor) => _flavor.name === name),
+    }));
+    return result
+      .map((flavor) => {
+        const addon = catalog.addons.find(
+          (_addon) => _addon.planCode === flavor.planCodes.hourly,
+        );
+        const addonMonthly = catalog.addons.find(
+          (_addon) => _addon.planCode === flavor.planCodes.monthly,
+        );
+        const plan = availability.plans?.find(
+          (_plan) => _plan.code === flavor.planCodes.hourly,
+        );
+
+        // Bugfix: Issue with selecting the correct monthly price in the US
+        // -----------------------------------------------------------
+        // Context: Normally, we take the first element of the `pricings` array
+        // to determine the monthly price. However, for US offers, this first
+        // value may not be correct. Instead, we need to specifically look
+        // for the entry containing 'renew' in `capacities`.
+        //
+        // Solution:
+        // - We first search the `pricings` array for an entry where `capacities`
+        //   includes 'renew' and return it if found.
+        // - If no such entry exists, we fallback to the first element of `pricings` (if available).
+        const priceMonthly =
+          addonMonthly &&
+          (addonMonthly.pricings.find((p) => p.capacities.includes('renew')) ||
+            addonMonthly.pricings[0]);
+
+        return {
+          ...flavor,
+          blobs: addon?.blobs,
+          compatibility: Object.values(DeploymentMode).reduce(
+            (acc, value) => ({
+              ...acc,
+              [value]: plan?.regions?.some((_region) => _region.type === value),
+            }),
+            {},
+          ) as Record<DeploymentMode, boolean>,
+
+          pricingsHourly: addon?.pricings?.[0],
+          pricingsMonthly: priceMonthly,
+          isNew: addon?.blobs.tags.includes('is_new'),
+          flavorCategory: FLAVOR_CATEGORIES.find((cat) =>
+            cat.pattern.test(flavor.type),
+          )?.category,
+          isFlex: /flex$/.test(flavor.name),
+          isLegacy: /eg|sp|hg|vps-ssd/.test(flavor.name),
+          hasEnoughQuota: hasEnoughQuota(flavor, quota[0]),
+        };
+      })
+      .sort((a, b) => {
+        const aGroup = Number((a.name.match(/[0-9]+/) || [])[0]);
+        const bGroup = Number((b.name.match(/[0-9]+/) || [])[0]);
+        const aRank = Number((a.name.match(/-([^-]+)$/) || [])[1]);
+        const bRank = Number((b.name.match(/-([^-]+)$/) || [])[1]);
```

Comment:
Can you move the match call inside the map method ? This would avoid a lot of calls